django/utils/dateparse.py

```diff
@@ -128,3 +141,12 @@ def parse_duration(value):
             kw['microseconds'] = '-' + kw['microseconds']
         kw = {k: float(v) for k, v in kw.items() if v is not None}
         return sign * datetime.timedelta(**kw)
+    match = postgres_interval_re.match(value)
+    if match:
+        kw = match.groupdict()
```

Comment:
Did you consider trying to factor out the common code between the two `if match` branches? Not sure if that would help or hurt readability.